server/src/main/java/org/elasticsearch/features/FeatureService.java

```diff
@@ -56,7 +56,7 @@ public FeatureService(List<? extends FeatureSpecification> specs) {
                     );
                 }
 
-                if (hfe.getValue().onOrAfter(CLUSTER_FEATURES_ADDED_VERSION)) {
+                if (hfe.getValue().after(CLUSTER_FEATURES_ADDED_VERSION)) {
```

Comment:
This is ok - feature API was introduced in the middle of 8.12, so version checks were added using 8.12 before feature API was available, which should correctly be migrated to historical features.